core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -1066,11 +1186,39 @@ object Chunk
 
     /** Prepends a chunk to the start of this chunk queue. */
     def +:[O2 >: O](c: Chunk[O2]): Queue[O2] =
-      if (c.isEmpty) this else new Queue(c +: chunks, c.size + size)
+      if (c.isEmpty) this
+      else
+        c match {
```

Comment:
```suggestion
      if (c.isEmpty) this
      else if (this.isEmpty) c
      else
        c match {
```
and the same recommendation for `:+` down below.

Reply:
I think that `++` can simply call to this method (i.e. `+:`) because they both seem doing the same thing, only the return type is different (`++` returns `Chunk` whereas `+:` returns `Queue`).

Reply:
@satorg I made the changes in a slightly different way, in order to minimize the amount of allocations.